test/http/doc_examples.cpp

```diff
@@ -307,6 +311,50 @@ class doc_http_samples_test
         BEAST_EXPECTS(! ec, ec.message());
     }
 
+    void
+    doConstAndMutableBody()
+    {
+        test::pipe c{ios_};
+
+        std::array<char, 15> const body {{"Hello, world!\n"}};
+        {
+            request<const_body<std::array<char, 15>>> req;
+            req.version = 11;
+            req.method(verb::put);
+            req.target("/");
+            req.body = body;
+            req.prepare_payload();
+            write(c.client, req);
```

Comment:
This is interesting because note that you are writing the null terminator into the body. That's one of the problems with using `std::array` as the body type, its always the same size.